plugin/src/main/java/com/denizenscript/denizen/events/block/BlockDispensesScriptEvent.java

```diff
@@ -36,6 +36,13 @@ public class BlockDispensesScriptEvent extends BukkitScriptEvent implements List
 
     public BlockDispensesScriptEvent() {
         registerCouldMatcher("<block> dispenses <item>");
+        this.<BlockDispensesScriptEvent, LocationTag>registerDetermination(null, LocationTag.class, (evt, context, value) -> {
+            evt.event.setVelocity(value.toVector());
+        });
+        this.<BlockDispensesScriptEvent, ItemTag>registerDetermination(null, ItemTag.class, (evt, context, value) -> {
+            evt.item = value;
+            evt.event.setItem(item.getItemStack());
+        });

```

Comment:
For stuff like this should prefer just getting the item directly from the event in the context instead of updating the field - this works within Denizen, but not if another plugin changes the item after the field was already set.
Can keep the field for use in `matches` for performance reasons, but `getContext` should ideally get the updated item every time.